src/api/app/models/user_session.rb

```diff
@@ -0,0 +1,29 @@
+class UserSession
+  class FailedAuthentication < StandardError; end
+  class DisabledAccount < StandardError; end
+
+  # Returns the user storaged in current Thread
+  #
+  # @return [User|nil] user for current session or nil if none
+  def self.current
+    Thread.current[:user]
+  end
+
+  # Storages the user in current Thread
+  #
+  # @param [User|nil]
+  #
+  # @raise [FailedAuthentication] if not user given
```

Comment:
Again, this is evident! There is no different between reading the comments and the code, then the comments can be removed :wink: